cdap-common/src/main/java/io/cdap/cdap/common/guice/DFSLocationModule.java

```diff
@@ -66,16 +73,28 @@ private LocationFactoryProvider(CConfiguration cConf, Configuration hConf,
     @Override
     public LocationFactory get() {
       String namespace = cConf.get(Constants.CFG_HDFS_NAMESPACE);
-      LOG.debug("HDFS namespace is {}", namespace);
+      LOG.debug("Location namespace is {}", namespace);
+
+      LocationFactory lf;
 
       // This FileContextLocationFactory supports multiple users from the same process.
       // It is used when security is enabled, which in turn impersonation could occur.
       if (UserGroupInformation.isSecurityEnabled()) {
-        return new FileContextLocationFactory(hConf, namespace);
+        lf = new FileContextLocationFactory(hConf, namespace);
+      } else {
+        // In non hadoop secure mode, use the static file context, which operates as single user.
+        lf = new InsecureFileContextLocationFactory(hConf, namespace, staticFileContextProvider.get());
       }
 
-      // In non hadoop secure mode, use the static file context, which operates as single user.
-      return new InsecureFileContextLocationFactory(hConf, namespace, staticFileContextProvider.get());
+      String locationCachePath = cConf.get(Constants.LOCATION_CACHE_PATH);
+      Path cachePath = Strings.isNullOrEmpty(locationCachePath) ? null : Paths.get(locationCachePath).toAbsolutePath();
```

Comment:
would be a little simpler to return lf if locationCachePath is null before this line